File: apollo-api/src/main/java/com/apollographql/apollo/api/internal/Function.java
package com.apollographql.apollo.api.internal;

import org.jetbrains.annotations.NotNull;

public interface Function<T, R> {
  /**
   * Apply some calculation to the input value and return some other value.
   *
   * @param t the input value
   * @return the output value
   */
  @NotNull
  R apply(@NotNull T t);
}

File: deprecated-apollo-runtime/src/main/java/com/apollographql/apollo3/api/internal/Function.kt
package com.apollographql.apollo3.api.internal

@Deprecated("This shouldn't be used anymore with Kotlin APIs")
interface Function<T, R> {
  /**
   * Apply some calculation to the input value and return some other value.
   *
   * @param t the input value
   * @return the output value
   */
  fun apply(t: T): R
}